File: java/src/main/java/controller/PaymentsController.java
package controller;

import com.checkout.CheckoutApiException;
import com.checkout.CheckoutApiImpl;
import com.checkout.CheckoutAuthorizationException;
import com.checkout.CheckoutSdk;
import com.checkout.Environment;
import com.checkout.common.Currency;
import com.checkout.payments.request.PaymentRequest;
import com.checkout.payments.request.source.RequestTokenSource;
import com.checkout.payments.response.PaymentResponse;
import com.checkout.tokens.CardTokenResponse;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import errors.ErrorResponse;
import org.eclipse.jetty.http.HttpStatus;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import spark.Request;
import spark.Response;

import java.util.concurrent.ExecutionException;

import static java.util.Objects.requireNonNull;

public class PaymentsController {

    private PaymentsController() {
    }

    private static final Logger LOG = LoggerFactory.getLogger(PaymentsController.class);
    private static final Gson serializer = new GsonBuilder().create();

    public static String pay(Request req, Response res) {

        CheckoutApiImpl checkoutSdk;
        try {
            checkoutSdk = CheckoutSdk.builder()
                .staticKeys()
                .secretKey(requireNonNull("sk_sbox_XXX"))
                .environment(Environment.SANDBOX)
                .build();
        } catch (CheckoutAuthorizationException e) {
            LOG.error("An error occurred while initializing Checkout SDK - {}", e.getMessage());
            res.status(400);
            return serializer.toJson(new ErrorResponse(400, e.getMessage()));
        }

        CardTokenResponse cardToken = serializer.fromJson(req.body(), CardTokenResponse.class);

        RequestTokenSource tokenSource = RequestTokenSource.builder()
            .token(cardToken.getToken())
            .build();

        PaymentRequest request = PaymentRequest.builder()
            .source(tokenSource)
            .amount(2499L)
            .currency(Currency.GBP)
            .processingChannelId("pc_XXX")
            .build();

        PaymentResponse response;
        try {
            response = checkoutSdk.paymentsClient().requestPayment(request).get();
        } catch (InterruptedException | ExecutionException e) {
            LOG.error("An error occurred while processing payment request - {}", e.getMessage());
            CheckoutApiException chkException = parseException(e);
            res.status(chkException.getHttpStatusCode());
            return serializer.toJson(new ErrorResponse(chkException.getHttpStatusCode(), e.getMessage()));
        }

        return serializer.toJson(response);
    }

    private static CheckoutApiException parseException(Exception e) {
        if (e.getCause() instanceof CheckoutApiException) {
            return (CheckoutApiException) e.getCause();
        }
        return new CheckoutApiException(HttpStatus.BAD_REQUEST_400, null, null);
    }
}

File: go/src/api/controllers/payments_controller.go
// github.com/checkout/checkout-sdk-go v1.0.5

package controllers

import (
	"encoding/json"
	"log"
	"net/http"

	"github.com/checkout/checkout-sdk-go"
	"github.com/checkout/checkout-sdk-go/common"
	"github.com/checkout/checkout-sdk-go/configuration"
	"github.com/checkout/checkout-sdk-go/errors"
	payments "github.com/checkout/checkout-sdk-go/payments/nas"
	"github.com/checkout/checkout-sdk-go/payments/nas/sources"
	"github.com/checkout/checkout-sdk-go/tokens"
)

type (
	PaymentsController struct{}
)

func NewPaymentsController() *PaymentsController {
	return &PaymentsController{}
}

func (c *PaymentsController) RequestPayment(w http.ResponseWriter, req *http.Request) {
	sdk, err := checkout.Builder().
		StaticKeys().
		WithEnvironment(configuration.Sandbox()).
		WithSecretKey("sk_sbox_XXX").
		Build()
	if err != nil {
		log.Printf("An error occurred while initializing Checkout SDK - %s", err.Error())
		http.Error(w, err.Error(), http.StatusBadRequest)
		return
	}

	var cardToken tokens.CardTokenResponse
	if err = json.NewDecoder(req.Body).Decode(&cardToken); err != nil {
		log.Printf("An error occurred while deserealizing card token - %s", err.Error())
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	tokenSource := sources.NewRequestTokenSource()
	tokenSource.Token = cardToken.Token

	paymentRequest := payments.PaymentRequest{
		Source:              tokenSource,
		Amount:              2499,
		Currency:            common.GBP,
		ProcessingChannelId: "pc_XXX",
	}

	response, err := sdk.Payments.RequestPayment(paymentRequest, nil)
	if err != nil {
		ckoErr := err.(errors.CheckoutAPIError)
		log.Printf("An error occurred while requesting payment - %s", ckoErr.Status)
		http.Error(w, ckoErr.Status, ckoErr.StatusCode)
		return
	}

	handleResponse(w, response.HttpMetadata.StatusCode, response)
}

func handleResponse(w http.ResponseWriter, status int, data interface{}) {
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(status)
	json.NewEncoder(w).Encode(data)
}
